controllers/automation/policyautomation_controller.go

```diff
@@ -199,6 +205,131 @@ func (r *PolicyAutomationReconciler) Reconcile(
+				originalStartTime, err := time.Parse(time.RFC3339, clusterEvent.AutomationStartTime)
+				if err != nil {
+					log.Error(err, "Failed to retrieve AutomationStartTime in ClustersWithEvent")
+					delete(eventMap, clusterName)
+				}
+
+				preEventTime, err := time.Parse(time.RFC3339, clusterEvent.EventTime)
+				if err != nil {
+					log.Error(err, "Failed to retrieve EventTime in ClustersWithEvent")
+					delete(eventMap, clusterName)
+				}
+
+				// The time that delayAfterRunSeconds setting expires
+				delayUntil := originalStartTime.Add(time.Duration(delayAfterRunSeconds) * time.Second)
+
+				// The policy is non-compliant with the target cluster
+				if targetListMap[clusterName] {
+					// Policy status changed from non-compliant to compliant
+					// then back to non-compliant during the delay period
+					if delayAfterRunSeconds > 0 && preEventTime.After(originalStartTime) {
+						if now.After(delayUntil) {
+							// The delay period passed so remove the previous event
+							delete(eventMap, clusterName)
+							// Add the cluster name to create a new ansible job
+							trimmedTargetList = append(trimmedTargetList, clusterName)
+						} else {
+							requeueFlag = true
+							// Within the delay period and use the earliest requeueDuration to requeue
+							if (requeueDuration == 0) || (requeueDuration > int(delayUntil.Sub(now)+1)) {
+								requeueDuration = int(delayUntil.Sub(now) + 1)
+							}
+							// keep the event and update eventTime
+							clusterEvent.EventTime = nowStr
+							// new event from compliant to non-compliant
+							eventMap[clusterName] = clusterEvent
+						}
+					} // Otherwise, the policy keeps non-compliant since originalStartTime, do nothing
+				} else { // The policy is compliant with the target cluster
+					if delayAfterRunSeconds > 0 && now.Before(delayUntil) {
+						// Within the delay period, keep the event and update eventTime
+						clusterEvent.EventTime = nowStr
+						// new event from non-compliant to compliant
+						eventMap[clusterName] = clusterEvent
+					} else { // No delay period or it is expired, remove the event
+						delete(eventMap, clusterName)
+					}
+				}
+			}
+
+			for _, clusterName := range targetList {
+				if _, ok := eventMap[clusterName]; !ok {
+					// Add the non-compliant clusters without previous automation event
+					trimmedTargetList = append(trimmedTargetList, clusterName)
+				}
+			}
+
+			if len(trimmedTargetList) > 0 {
+				log.Info("Creating An Ansible job", "trimmedTargetList", trimmedTargetList)
+				err = common.CreateAnsibleJob(
+					policyAutomation,
+					r.DynamicClient,
+					string(policyv1beta1.EveryEvent),
+					trimmedTargetList,
+				)
+
+				if err != nil {
+					log.Error(err, "Failed to create the Ansible job")
+
+					return reconcile.Result{}, err
+				}
+
+				automationStartTimeStr := time.Now().UTC().Format(time.RFC3339)
+
+				for _, clusterName := range trimmedTargetList {
```

Comment:
What about requeuing when a new target is added to the list that wasn't in `eventMap`?

Reply:
@mprahl So when a new target is added to the list that wasn't in `eventMap`, we created an ansible job immediately, any reason for `requeuing`.

Reply:
> What about requeuing when a new target is added to the list that wasn't in `eventMap`?

Yes, you are correct. Need to re-queue when a new target is added to the list that wasn't in `eventMap`, since EventHook is only for `noncompliant`, not for `compliant`

Reply:
Actually, now that I'm thinking about it, maybe we don't need to since there is no action that can be taken unless it goes to compliant. That was my mistake.